packages/charts/react-charts/library/src/components/HorizontalBarChart/HorizontalBarChart.tsx

```diff
@@ -109,39 +110,39 @@ export const HorizontalBarChart: React.FunctionComponent<HorizontalBarChartProps
   };
 
   function _createLegends(chartProps: ChartProps[]): JSX.Element {
-    const legendItems: Legend[] = [];
-    chartProps.forEach((point: ChartProps) => {
-      point.chartData!.forEach((dataPoint: ChartDataPoint) => {
-        const color: string = dataPoint.color!;
-        // mapping data to the format Legends component needs
-        const legendItem: Legend = {
-          title: dataPoint.legend!,
-          color,
-          action: () => {
-            if (selectedLegend === dataPoint.legend) {
-              setSelectedLegend('');
-            } else {
-              setSelectedLegend(dataPoint.legend!);
-            }
-          },
-          hoverAction: () => {
-            _handleChartMouseLeave();
-            setActiveLegend(dataPoint.legend!);
-          },
-          onMouseOutAction: () => {
-            setActiveLegend('');
-          },
-        };
-        legendItems.push(legendItem);
-      });
-    });
-    const legends = (
+    const legendItems: Legend[] = chartProps.flatMap(
+      point =>
+        point.chartData?.map((dataPoint): Legend => {
+          const legend = dataPoint.legend ?? '';
+          const color = dataPoint.color ?? '';
+          const annotation = dataPoint.annotation?.() ?? null;
+
+          return {
+            title: legend,
+            color,
+            action: () => setSelectedLegend(selectedLegend === legend ? '' : legend),
+            hoverAction: () => {
+              _handleChartMouseLeave();
+              setActiveLegend(legend);
+            },
+            onMouseOutAction: () => setActiveLegend(''),
+            legendAnnotation: () => annotation,
```

Comment:
> legendAnnotation: () => annotation, [](http://example.com/codeflow?start=11&length=36)

annotation should be a pass through props. Why is any chanhe needed in horizontal bar chart